runtime/tests/interpreter/capability_test.go

```diff
@@ -319,6 +343,12 @@ func TestInterpretCapability_borrow(t *testing.T) {
 			_, err := inter.Invoke("loop")
 			require.Error(t, err)
 
+			require.IsType(t,
+				interpreter.Error{},
+				err,
+			)
+			err = err.(interpreter.Error).Unwrap()
+
 			require.IsType(t, interpreter.CyclicLinkError{}, err)
```

Comment:
I think you can use `errors.As` to avoid the explicit unwrapping step in all these checks. `As` recursively unwraps the error and checks whether it can be assigned to a given type which should work here. Something like:
```go
require.True(t, errors.As(err, &interpreter.CyclicLinkError{})
```

Reply:
Great suggestions! I wanted to have a better way, this is great 👌 Refactored as many cases as I could in 98771cb.

I checked tesitify and they'll have a `ErrorAs` assertion function soon (https://github.com/stretchr/testify/commit/95a9d909e98735cd8211dfc5cbbb6b8b0b665915), so for now I added a helper that does the same and added a TODO to replace it once testify has released their function.